src/pages/chapter/profile.tsx

```diff
@@ -5,41 +5,143 @@ import {
 import { states, countries } from "src/utils/constants";
-import { NonprofitStage, ProjectType } from "src/utils/types";
+import {
+  UserUpdate,
+  ChapterUpdate,
+  MaintenanceType,
+  Chapter,
+} from "src/utils/types";
 
-const requiredStages = [NonprofitStage.IN_PROGRESS, NonprofitStage.COMPLETE];
+interface FormData {
+  name: string;
+  phoneNumber: string;
+  chapterName: string;
+  contact: string;
+  street: string;
+  city: string;
+  state: string;
+  zipCode: number;
+  country: string;
+  website: string;
+  facebook: string;
+  instagram: string;
+  maintenanceTypes: string[];
+  maintenancePeriod: number;
+}
+
+interface Contact {
+  id: string;
+  name: string;
+  email: string;
+}
 
 function ChapterProfilePage() {
-  const [name, setName] = useState("");
-  const [calendly, setCalendly] = useState("");
-  const [phoneNumber, setPhoneNumber] = useState("");
+  const {
+    control,
+    handleSubmit,
+    register,
+    formState: { errors, isSubmitting },
+    reset,
+  } = useForm<FormData>();
+
+  const [contactList, setContactList] = useState<Contact[]>([]);
+
+  useEffect(() => {
+    async function preloadFields() {
+      const user = await getUserProfile();
+      const chapter = user.chapter as Chapter;
 
-  const [chapterName, setChapterName] = useState("");
-  const [street, setStreet] = useState("");
-  const [city, setCity] = useState("");
-  const [state, setState] = useState(states[0]);
-  const [zip, setZip] = useState("");
-  const [country, setCountry] = useState(countries[0]);
-  const [chapterEmail, setChapterEmail] = useState("");
-  const [projectLimit, setProjectLimit] = useState("1");
-  const [website, setWebsite] = useState("");
-  const [facebook, setFacebook] = useState("");
-  const [instagram, setInstagram] = useState("");
-  const [chapterProjects, setChapterProjects] = useState<ProjectType[]>([]);
+      const contacts = await getChapterUsers();
+      setContactList(
+        contacts.map((user) => {
+          return {
+            id: user._id.toString(),
+            name: user.name,
+            email: user.email,
+          };
+        })
+      );
 
-  const [projectProcess, setProjectProcess] = useState<NonprofitStage[]>([
-    ...requiredStages,
-  ]);
+      reset({
+        name: user.name,
+        phoneNumber: user.phoneNum ?? "",
+        chapterName: chapter.name,
+        contact: chapter.contact.toString(),
+        street: chapter.address.street,
+        city: chapter.address.city,
+        state: chapter.address.state,
+        zipCode: chapter.address.zipCode,
+        country: chapter.address.country,
+        website: chapter.website ?? "",
+        facebook: chapter.facebook,
+        instagram: chapter.instagram,
+        maintenanceTypes: chapter.maintenanceTypes,
+        maintenancePeriod: chapter.maintenancePeriod,
+      });
+    }
+
+    preloadFields().catch((error: Error) => {
+      console.log(error.message);
```

Comment:
Use the showError function to display an error here instead of logging it so that the user is aware of the problem.